src/assemblyItemsBrowser.js

```diff
@@ -192,13 +198,58 @@ module.exports = React.createClass({
 		}
 		
 		var currentAddress = this.state.currentAddress
-		if (props.vmTrace[vmTraceIndex].address) // there's a new context
-		{
-			var stack = this.state.callStack[vmTraceIndex]
-			currentAddress = stack[stack.length - 1]
-			this.resolveAddress(currentAddress, props)
-			Object.assign(stateChanges, { "currentAddress": currentAddress })
-		}
+        var addressIndex = this.shouldUpdateStateProperty("address", vmTraceIndex, previousIndex, props.vmTrace)
+        if (addressIndex > -1)
+        {
+            if (this.state.currentSelected === -1) // beginning of the trace
+            {
+                currentAddress = props.vmTrace[addressIndex].address
+                if (props.transaction.to === null) // contract creation
+                {
+                    this.cacheCode(currentAddress, props.transaction.input)
```

Comment:
Ok but then `currentAddress` is perhaps a misleading name. I think we should create a proper module that only performs "code fetching" and adds a caching step in between, but that can be done in a later step.